src/v2/services/implementations/AssetsService.ts

```diff
@@ -26,33 +28,51 @@ export class AssetsService implements IAssetsService {
   }
 
   public async transferNativeAsset(param: ParamAssetTransfer): Promise<void> {
-    const transaction = await this.AssetsRepository.getNativeTransferCall(param);
-    const hash = await this.wallet.signAndSend({
-      extrinsic: transaction,
-      senderAddress: param.senderAddress,
-      successMessage: param.successMessage,
-    });
-    param.finalizedCallback(String(hash));
+    const useableBalance = await this.AssetsRepository.getNativeBalance(param.senderAddress);
+    const isBalanceNotEnough =
+      Number(ethers.utils.formatUnits(useableBalance, 18)) -
+        Number(ethers.utils.formatUnits(param.amount, 18)) <
+      REQUIRED_MINIMUM_BALANCE;
+
+    if (isBalanceNotEnough) {
+      throw new Error(AlertMsg.MINIMUM_BALANCE);
+    } else {
+      const transaction = await this.AssetsRepository.getNativeTransferCall(param);
+      const hash = await this.wallet.signAndSend({
+        extrinsic: transaction,
+        senderAddress: param.senderAddress,
+        successMessage: param.successMessage,
+      });
+      param.finalizedCallback(String(hash));
+    }
   }
 
   public async transferEvmAsset(param: ParamEvmTransfer): Promise<void> {
     const provider = getEvmProvider(this.currentWallet as any);
     const web3 = new Web3(provider as any);
-    const rawTx = await this.AssetsRepository.getEvmTransferData({
-      param,
-      web3,
-    });
 
-    const isErc20 = !!rawTx.data;
+    const balWei = await web3.eth.getBalance(param.senderAddress);
+    const useableBalance = Number(ethers.utils.formatUnits(balWei, 18));
```

Comment:
And here too
https://github.com/AstarNetwork/astar-apps/pull/1116/files#r1448571612